### PR TITLE
DefeatDungeonBossQuest for Surprise Togepi and Parasect from Manaphy questline

### PR DESCRIPTION
**Pull request by RegisCoaxans**
I added DefeatDungeonBossQuest, which, obviously, requires the player to defeat a specific boss in a dungeon, both trainer or wild pokémon work !
As an introduction, I fixed Surprise Togepi passively ruining its questline in BF.

### CODE REVIEW

**Review by osherwott — COMMENTED** [review]
Could you add this requirement to the Parasect in the Manaphy Quest and subsequently remove the extra requirements for the regular Eterna Forest bosses?

**Comment by RegisCoaxans** [comment]
Parasect is done.
Any other occurrence of this configuration that you know ?

**Comment by osherwott** [comment]
There are a few questionable dungeon clears that could benefit from this, but completely replacing questline steps is best left for another time, since it's a matter of content rather than code cleanup. As far as this PR's concerned, all's good 👍

File: src/scripts/quests/questTypes/DefeatDungeonBossQuest.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+        // Target is undefined until onLoad is called. TmpTarget keeps track of raw data.
+        this.focus = ko.pureComputed(() => {
+            const enemyPokemon = Battle.enemyPokemon();
+            const defeatedBoss = DungeonRunner.defeatedBoss();
+            if (enemyPokemon) {
+                if (enemyPokemon.name === this.target
+                    && DungeonRunner.dungeon.name === this.dungeon
+                    && App.game.gameState === GameConstants.GameState.dungeon) {
+                    this.targetEncountered = true;
```
Comment (Jaaslet) [inline src/scripts/quests/questTypes/DefeatDungeonBossQuest.ts:23]: I don't like pure computed setting other values. Those should be pure computed too

File: src/scripts/quests/questTypes/DefeatDungeonBossQuest.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
+class DefeatDungeonBossQuest extends Quest implements QuestInterface {
+    initialValue?: number;
+    customReward?: () => void;
+    target: PokemonNameType;
+    targetEncountered: boolean;
+
+    constructor(public dungeon: string, private tmpTarget : string, reward: (() => void) | number = 0, description?: string, onLoad?: (() => void)) {
```
Comment (Jaaslet) [inline src/scripts/quests/questTypes/DefeatDungeonBossQuest.ts:7]: `tmpTarget` is not very tmp, when we keep it around

**Review by Jaaslet — CHANGES_REQUESTED** [review]

**Review by Jaaslet — APPROVED** [review]
Looks good. Thanks for the PR

**Pull request merged**